starboard/nplb/player_get_audio_configuration_test.cc

```diff
@@ -0,0 +1,198 @@
+    const int kMaxNumberOfConfigurations = 32;
+    for (int index = 0;; index++) {
+      ASSERT_LT(index, kMaxNumberOfConfigurations);
+      SbMediaAudioConfiguration configuration;
+      if (SbPlayerGetAudioConfiguration(player_fixture.GetPlayer(), index,
+                                        &configuration)) {
+        configurations->push_back(configuration);
+      } else {
+        return;
+      }
+    }
+  }
+};
+
+TEST_P(SbPlayerGetAudioConfigurationTest, SunnyDay) {
+  const int kSamplesToWrite = 8;
+
+  SbPlayerTestFixture player_fixture(GetParam());
+  if (HasFatalFailure()) {
+    return;
+  }
+
+  // It's ok to not return any configuration at this point.
+  std::vector<SbMediaAudioConfiguration> initial_configs;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &initial_configs));
+
+  GroupedSamples samples;
+  if (player_fixture.HasAudio()) {
+    samples.AddAudioSamplesWithEOS(0, kSamplesToWrite);
+  }
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamplesWithEOS(0, kSamplesToWrite);
+  }
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  // The audio configurations must be available after `kSbPlayerStatePresenting`
+  // has been received.
+  std::vector<SbMediaAudioConfiguration> configs_after_presenting;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_presenting));
+
+  if (player_fixture.HasAudio()) {
+    ASSERT_FALSE(configs_after_presenting.empty());
+  }
+
+  // Once at least one audio configurations are returned, the return values
+  // shouldn't change during the life time of |player|.
+  if (!initial_configs.empty()) {
+    ASSERT_EQ(initial_configs, configs_after_presenting);
+  }
+
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerEndOfStream());
+
+  std::vector<SbMediaAudioConfiguration> configs_after_end;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_end));
+  ASSERT_EQ(configs_after_presenting, configs_after_end);
+}
+
+TEST_P(SbPlayerGetAudioConfigurationTest, NoInput) {
+  SbPlayerTestFixture player_fixture(GetParam());
+  if (HasFatalFailure()) {
+    return;
+  }
+
+  std::vector<SbMediaAudioConfiguration> initial_configs;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &initial_configs));
+
+  GroupedSamples samples;
+  if (player_fixture.HasAudio()) {
+    samples.AddAudioSamplesWithEOS(0, 0);
+  }
+  if (player_fixture.HasVideo()) {
+    samples.AddVideoSamplesWithEOS(0, 0);
+  }
+  ASSERT_NO_FATAL_FAILURE(player_fixture.Write(samples));
+  ASSERT_NO_FATAL_FAILURE(player_fixture.WaitForPlayerPresenting());
+
+  std::vector<SbMediaAudioConfiguration> configs_after_presenting;
+  ASSERT_NO_FATAL_FAILURE(
+      ReadAudioConfigurations(player_fixture, &configs_after_presenting));
+
+  if (player_fixture.HasAudio()) {
```

Comment:
It's my mistake that failed to take this situation into account.  I believe it's valid for the implementation to not have the audio output ready under this situation.  Please refine the comment of `SbPlayerGetAudioConfiguration()`, and ensure my implementation in SbPlayerPipeline works in this case.  Thank you!